### DISCUSSION TITLE
LilyGO V1.6 schematics

### DISCUSSION THREAD

**Discussion by faragher** [markqvist/Reticulum#383] [Show and tell]
For reference: Many of us end up getting the LilyGO hardware for an RNode pre-configured as the Paxcounter, and while I was poking around, I noticed they have the schematic available for those of you working on your own hardware or working on firmware.

It's available in other places, but I figured I'd mention it here.

https://github.com/LilyGO/ESP32-Paxcounter/blob/master/T3_V1.6.pdf

**Comment by githubques** [comment]
A few LilyGo related questions:

1) Best places online to buy?

In anticipation of LilyGo's serving as stand alone RNS repeaters in the future to spread out a node network's reach:

2) Best batteries to use and where to buy?

3) Best solar cells to use and where to buy?

4) Decent to Best antennas to use and where to buy?

**Reply by faragher** [reply]
Can't answer those questions.

I bought my test units from Amazon for customer protections and fast delivery. You can order them from the company for like 5 USD cheaper, but I don't know their shipping terms.

Batteries are their own large conversation. The RNode cases have stated battery capacities, just make sure the polarity is correct. I can't recommend a style, since bought the wrong connector and had to replace the connector manually.

I don't know solar cells.

Antennas are generally situation specific, but I've had good luck with the Low Power Lab dipole. No housing and an optional connector, but dirt cheap for tuned dipoles.

**Reply by githubques** [reply]
Thanks for the tip about the Low Power Lab dipole. I'll try to pick up a few in addition to the antenna that comes with the LilyGo.

Found a site (other than Amazon) to source the LilyGo and a couple 16550 batteries for my existing ones. Will need to find a source for 5W solar cells.

**Comment by cobraPA** [comment]
Note that the RNode is not running the network layer today, so using it as a repeater requires quite a bit of work, or simpler would be a pair of RNode's (potentially) and something running the network/Reticulum.  I guess one RNode and a Reticulum device would work as well.

**Reply by githubques** [reply]
Since RNS can carry it's signal by internet, wifi, bluetooth, and ham radio signal... 

1) Is it also possible to carry their signal through Meshtastic's network which has repeaters and uses the same LilyGo's on 915 mHz?

2) Since you can partition a cpu to be half Windows OS and half Linux OS, is it possible to partition a LilyGo to be half Meshtastic and half Reticulum?

Given Reticulum is still under development and Meshtastic is operational, would it not be a good idea to build up a Meshtastic network first to get the infrastructure in place in an area and then add to/replace the Meshtastic network with RNS as it comes fully online?

I hope there's an "add to" scenario since replacing means going to every existing LilyGo and manually replacing the Meshtastic OS with a new RNS one (barring an easier OS replacement option).

**Reply by faragher** [reply]
I'll have to leave specifics and final explanations to Mark but,

1. Data is data, writing an interface for Reticulum could make that happen. See below.
2. No. You might be able to use an RNode to act as a modem to run Meshtastic on a computer, but you'd have to process each packet, determine which protocol it belonged to, and handle it appropriately. Basically you'd have to write a hybrid firmware that would take up far more development time than it was worth.

The issue here is you can't just run two systems side by side and think "eventually we'll switch over." You build infrastructure or you use an existing one; half measures are universally counterproductive. If we worked on top of Meshtastic, then people would build on top of it and never want to move to RNode, and we're back to the discussion we had about TCP bridging being a terrible idea.

Bottom line, it's best to run purpose built infrastructure rather than use something that doesn't share all the advantages the current system provides.

But once again, connecting via a RNS Interface is certainly possible. Data is data. It's just wasteful and suboptimal. Reticulum is a network stack, not an app on top of another network stack.

**Reply by githubques** [reply]
_"I guess one RNode and a Reticulum device would work as well."_

**1) And what's the cheapest/easiest way to make a pseudo-repeater?**

RpI Pico's loaded up with Ubuntu running RNS with a T-Beam and battery connected to it inside a protected case on a window sill?

Is there something smaller and cheaper than a Pico that can run a microkernel OS with RNS installed on it to create a repeater?

**2) For non-Android users, what's the easiest/cheapest way to make a hand held RNS/SideBand device that has a screen and touch pad like the LILYGO T-Deck Microcontrollers?**

There was mention that RNS had been successfully loaded on certain Kindles. Are there older kindle versions that can be bought cheap and loaded up with RNS to make portable RNS devices?

It would be nice to setup and toy around with a small network even if only as an emergency back up communications system.

**Reply by faragher** [reply]
I can't speak to question 1. I've only tried a small selection of relatively powerful hardware (Pis, Libre SBCs, etc.)

2: If you're going to buy new hardware, why not go android? It's dirt cheap for secondhand phones and if you don't like Google you can do some research, buy something you can root, and install something like LineageOS. The upcoming version of Sideband does what you'll need and in the future should cover practically all use cases.

**Reply by githubques** [reply]
@faragher

You had me at "dirt cheap"

I looked into used Androids awhile back and found pricing for used ranging from $100-400 which isn't that cheap, especially if looking to buy a few.

Looked again based on your recommendation and found some in the $30-50 range.

**1) Question is, even when they list Android as their operating system, are they the right ones?**

I know Mark did some backwards compatibility for older Androids, but don't remember what the cut off is or where to find a list of what works like he has posted on unsigned.io for the best RNode's to use like LilyGo's.

**2) Do you recommend LineageOS over GrapheneOS?**

I don't plan on adding a carrier for these phones since they'll be purely for RNS/SB testing. Not to mention some of the listed phones are likely no longer supported by many carriers.

So the venn diagram requires overlap between Price, RNS/SB, and Lineage/Graphene compatibility.

If naming specific manufacturers and models is too vast to do, is there a common spec the phone needs to have for RNS/SB? From there I can see if that limited selection supports Lineage and/or Graphene.

**Reply by faragher** [reply]
Remember to check at cell phone shops, too. Of course they want to finance you a new phone, with insurance and blah blah blah, but if you get pre-paid service, they often have a cheap phone, like an LG, so discounted it's ridiculous. Just buy the phone, pay for a month, and never reload it. You're not even violating their terms of service, just taking advantage of a good deal.

I have an Android 8 and an Android 9, both with Snapdragon 630 chipsets. https://nanoreview.net/en/soc/qualcomm-snapdragon-630 They are lackluster industrial chips and basically anything made in the last five years, and certainly any flagship phones from the last 10, will blow them out of the water. Sideband is slow to boot (should change in the next update) but runs fine once open. 

I only used LineageOS as a stepping stone to Kali Nethunter. Any Android OS with access to FDroid should run Sideband just fine (and let us know if it doesn't!). Even without FDroid you can always load the APK directly, I just find it convenient. 

That being said, if you pull the SIM and stay off WiFi, you don't even need to root the phone unless you really, really hate Google.

**Reply by githubques** [reply]
_"I have an Android 8 and an Android 9, both with Snapdragon 630 chipsets."_

Looks like I found a deal on Coolpad Legacy running Android version 9 (Pie), however the specs list it as having Qualcomm Snapdragon 450. Is 630 the floor or will 450 do the trick?

If the Coolpad is a no go, I'll expand the search to the LG line of phones you mentioned.

_"That being said, if you pull the SIM and stay off WiFi, you don't even need to root the phone unless you really, really hate Google."_

I like playing around and if loading Lineage and/or Graphene gets rid of standard Google while still allowing RNS/SB to run as intended, I'll consider that a bonus.

**Reply by faragher** [reply]
There's nothing special about the 630, I just have two and they function fine. Actually very well since the last update. 

https://nanoreview.net/en/soc-compare/qualcomm-snapdragon-630-vs-qualcomm-snapdragon-450

It's a big difference but not impossible. If you're willing to take the chance, money wise, I'd consider it. Please let us know how it works either way!

I love playing around with tech. I sadly bricked a tablet that way, but it was something I got for free. Speaking of, ask around at **non-corporate** cell phone shops, schools, etc. A lot of places have junk devices they haven't gotten around to taking somewhere, and if even half work, you're golden. Garbage tablet anyway.

**Reply by githubques** [reply]
I'll go for it with a 450 or two just see if it works like you say.

1) It says it works with Sprint only as the carrier. I won't be bothering with a carrier, but my question is will I be able to load Lineage OS (Graphene is Pixel only right now) and RNS/SB through FDroid or APK as soon as I get it or will I first have to set up a prepaid account with Sprint for at least a month?

If prepaid is the necessary, that complicates things since I'm planning on getting a few phones to test a mesh network and those costs can add up offsetting the "dirt cheap" price of each phone.

2) About two years ago, I came across a half dozen discarded droids, but all except one were too old for RNS. The one that looked like it would work was locked out because I didn't know the password and ended up giving it back. I've got another in my possession from someone who inherited it, but wasn't given the password. Is there a way to get into/wipe and reinstall these?

**Reply by kriscrossapplesauce2004** [reply]
You need to boot the phone into recovery mode; hold down the volume, power and home button until you get what looks like a terminal


And no, you don't need to prepay
